client/src/components/project-selection/ProjectSelector.tsx

```diff
@@ -0,0 +1,165 @@
+    GetChangeBetweenProjects(projectClick); //api call
+    // setProjects((projects) => {
+    //   const updatedProjects = projects.map((project) => ({
+    //     ...project,
+    //     isOpen: project.name === projectClick.target.value,
+    //   }));
+
+    //   console.log(updatedProjects); // Log for checking
+    //   return updatedProjects;
+    // });
+  };
+
+  const handleDeleteProject = (e: React.MouseEvent, projectName: string) => {
+    e.stopPropagation();
+    DeleteProject(e, projectName); //api call
+
+    // console.log("delete project", projectName);
+    // setProjects((prevProjects) =>
+    //   prevProjects.filter((project) => project.name !== projectName)
+    // );
+  };
+
+  //Add new project functions
+  const handleNewProjectSave = () => {
+    // Add new project logic here
+    if (
+      newProjectName === "" ||
+      projects.find((project) => project.name === newProjectName)
+    ) {
+      showSnackbar("Name cannot be empty or already exists", "error");
+      return;
+    }
+    PostAddNewProject(newProjectName); //api call
+
+    // const newProject: ProjectType = {
+    //   name: newProjectName,
+    //   id: String(projects[projects.length - 1].id + 1),
+    //   isOpen: true,
+    // };
+
+    // setProjects((prevProjects) => [...prevProjects, newProject]);
+    setNewProjectName("");
+    setNewProjectModalIsOpen(false);
+  };
+  const handleNewProjectCancel = () => {
+    setNewProjectName("");
+    setNewProjectModalIsOpen(false);
+  };
+
+  //Edit project functions
+  const handleEditProject = (e: React.MouseEvent, projectName: string) => {
+    e.stopPropagation();
+    setSelectedProjectName(projectName);
+    setEditedProjectName(projectName);
+    setEditProjectModalIsOpen(true);
+  };
+
+  const handleEditedProjectSave = () => {
+    if (
+      editedProjectName === "" ||
+      editedProjectName === selectedProjectName ||
+      projects.find((project) => project.name === editedProjectName)
+    ) {
+      showSnackbar("Name cannot be empty or the same", "error");
+      return;
+    }
+    PutEditProject(editedProjectName, selectedProjectName); //api call
+    // projects.map((project) => {
+    //   if (project.name === selectedProjectName) {
+    //     project.name = editedProjectName;
+    //   }
+    // });
+
+    setNewProjectName("");
+    setSelectedProjectName("");
+    setEditProjectModalIsOpen(false);
+  };
+
+  const handleEditedProjectCancel = () => {
+    setEditedProjectName("");
+    setSelectedProjectName("");
+    setEditProjectModalIsOpen(false);
+  };
+
+  return (
+    <div>
+      {snackBar}
+      <FormControl fullWidth size="small">
+        <InputLabel id="demo-simple-select-label ">Projects manager</InputLabel>
+        <Select
+          style={{ width: "200px" }}
+          labelId="demo-simple-select-label"
+          id="demo-simple-select"
+          value={
+            projects.find((project) => project.isOpen === true)?.name || ""
```

Comment:
Project selection should be managed in the frontend and not as a property from the backend.

Reply:
The `projects.find((project) => project.isOpen === true)?.name || ""` code is used to retrieve the name of the project with the `isOpen` property set to `true` in the local state (`projects`). If such a project is found, its name is used; otherwise, it defaults to an empty string.
Or did I not understand?

Reply:
`Projects` store the projects the came from the backend api call. 
You should not get from the backend this state. 
That's why you have defined:
`const [selectedProjectName, setSelectedProjectName] = useState("");`